source/extensions/filters/http/oauth2/config.cc

```diff
@@ -64,6 +64,14 @@ Http::FilterFactoryCb OAuth2Config::createFilterFactoryFromProtoTyped(
     throw EnvoyException("invalid HMAC secret configuration");
   }
 
+  if (proto_config.preserve_authorization_header() == true &&
+      proto_config.forward_bearer_token() == true) {
+    throw EnvoyException(
+        "invalid combination of forward_bearer_token and preserve_authorization_header "
+        "configuration. If forward_bearer_token is set to true, then "
+        "preserve_authorization_header must be false");
+  }
+
```

Comment:
Please add a test for this case.

Reply:
agree, can you give me some guidance, where is the best place for this kind of test? And maybe point me to an example were an config error is already tested. That would be really helpful

Reply:
@wbpcode

Reply:
https://github.com/envoyproxy/envoy/blob/b6b585b2baf468f5876349658c5a7d3424702810/test/extensions/filters/http/oauth2/config_test.cc#L150